clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -195,6 +200,30 @@ public class AzureComputeClient extends AzureBaseClient {
     instances
   }
 
+  Boolean waitForScaleSetHealthy(String resourceGroupName, String serverGroupName, long timeoutMillis) {
+    // check the scale set's health status, wait for the timeout return true when healthy, false if we hit the timeout
+
+    def instances = getServerGroupInstances(resourceGroupName, serverGroupName)
+    def now = System.nanoTime()
+    def currentTime = System.nanoTime()
+
+    while (currentTime - now < timeoutMillis * 1000000) {
+      def healthy = true
```

Comment:
```groovy
def healthy = !instances.any { it.healthState != HealthState.Up }
```

Note that `instances.every { it.healthState == HealthState.Up }` wouldn't be the same in the zero-instance case.